src/ProjectTemplates/test/BlazorWasmTemplateTest.cs

```diff
@@ -0,0 +1,257 @@
+            Assert.True(File.Exists(Path.Combine(publishDir, "service-worker.js")), "service-worker.js should be published");
+            Assert.True(File.Exists(Path.Combine(publishDir, "service-worker-assets.js")), "service-worker-assets.js should be published");
+
+            // We automatically append the SWAM version as a comment in the published service worker file
+            var serviceWorkerAssetsManifestContents = ReadFile(publishDir, "service-worker-assets.js");
+            var serviceWorkerContents = ReadFile(publishDir, "service-worker.js");
+
+            // Parse the "version": "..." value from the SWAM, and check it's in the service worker
+            var serviceWorkerAssetsManifestVersionMatch = new Regex(@"^\s*\""version\"":\s*(\""[^\""]+\"")", RegexOptions.Multiline)
+                .Match(serviceWorkerAssetsManifestContents);
+            Assert.True(serviceWorkerAssetsManifestVersionMatch.Success);
+            var serviceWorkerAssetsManifestVersionJson = serviceWorkerAssetsManifestVersionMatch.Groups[1].Captures[0].Value;
+            var serviceWorkerAssetsManifestVersion = JsonSerializer.Deserialize<string>(serviceWorkerAssetsManifestVersionJson);
+            Assert.True(serviceWorkerContents.Contains($"/* Manifest version: {serviceWorkerAssetsManifestVersion} */", StringComparison.Ordinal));
+        }
+
+        [ConditionalFact]
+        // LocalDB doesn't work on non Windows platforms
+        [OSSkipCondition(OperatingSystems.Linux | OperatingSystems.MacOSX)]
+        public Task BlazorWasmHostedTemplate_IndividualAuth_Works_WithLocalDB()
+            => BlazorWasmHostedTemplate_IndividualAuth_Works(true);
+
+        [Fact]
+        [SkipOnHelix("https://github.com/dotnet/aspnetcore/issues/30825", Queues = "All.OSX")]
+        public Task BlazorWasmHostedTemplate_IndividualAuth_Works_WithOutLocalDB()
+            => BlazorWasmHostedTemplate_IndividualAuth_Works(false);
+
+        private async Task<Project> CreateBuildPublishIndividualAuthProject(bool useLocalDb)
+        {
+            // Additional arguments are needed. See: https://github.com/dotnet/aspnetcore/issues/24278
+            Environment.SetEnvironmentVariable("EnableDefaultScopedCssItems", "true");
+
+            var project = await CreateBuildPublishAsync("blazorhostedindividual" + (useLocalDb ? "uld" : ""),
+                args: new[] { "--hosted", "-au", "Individual", useLocalDb ? "-uld" : "" });
+
+            var serverProject = GetSubProject(project, "Server", $"{project.ProjectName}.Server");
+
+            var serverProjectFileContents = ReadFile(serverProject.TemplateOutputDir, $"{serverProject.ProjectName}.csproj");
+            if (!useLocalDb)
+            {
+                Assert.Contains(".db", serverProjectFileContents);
+            }
+
+            var appSettings = ReadFile(serverProject.TemplateOutputDir, "appsettings.json");
+            var element = JsonSerializer.Deserialize<JsonElement>(appSettings);
```

Comment:
Any chance you could only use STJ? Most of the things this is doing should be achievable using a JsonDocument or using an actual type.

Reply:
I just lifted this from the existing template tests, do you want me to file a cleanup issue to switching both of these?